src/SearchBook.js

```diff
@@ -0,0 +1,90 @@
+import React from "react"
+import {Link} from "react-router-dom"
+import Book from "./Book"
+
+import * as BooksAPI from './BooksAPI'
+
+class SearchBook extends React.Component {
+  state = {
+    enteredText: '',
+    queriedBooks: [],
+    myBooks: []
+  }
+  componentDidMount() {
+    BooksAPI.getAll().then(myBooks => {
+      this.setState({ myBooks })
+    })
+  }
+  handleChange = (enteredText) => {
+    this.setState(() => {return {enteredText: enteredText}})
+    this.search_books(enteredText)
+  }
+  search_books = (serchText) => {
+    if(serchText) {
+      BooksAPI.search(serchText).then((books) => {
+        if(books && books.length) {
+          books = books.map(book => {
+            book.shelf = this.addShelf(book)
+            return book
+          })
+
+          this.setState(() => {
+            return {queriedBooks: books}
+          })
+        } else {
+          this.setState(() => {
+            return {queriedBooks: []}
+          })
+        }
+      })
+    } else {
+      this.setState({queriedBooks: [], enteredText: ''})
+    }
+  }
+  addShelf(book) {
+    let hasShelf = this.state.myBooks.filter(myBook => myBook.id === book.id)
+    return hasShelf.length ? "do_not_show" : "noneShelf"
+  }
+  changeShelf = (e, filteredBook) => {
+    const queriedBooks = this.state.queriedBooks
+    const shelf = e.target.value
+    filteredBook.shelf = e.target.value
+    this.setState({
+      queriedBooks
```

Comment:
Similar to another comment... above you set `queriedBooks` to what was in the state, and then here is setting the state to that variable without making any changes to it.

Reply:
removed